### PR TITLE
refactor: replace got() with fetch() in runner

### PR DESCRIPTION
**Pull request by ckerr**
Currently we use `fetch()` in modules/bot/ and modules/broker/tests, and `got()` in modules/runner/ .

This minor refactor PR changes the latter to avoid duplicate libraries.

### CODE REVIEW

**Comment by ckerr** [comment]
Adding @clavin as reviewer since runner is the one using got().

This isn't a big deal, so if you have a strong preference for got() that's fine. However if you don't, this PR lets us avoid depending on multiple HTTP modules

File: modules/bot/src/github-client.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import debug from 'debug';
 import { Context, Probot } from 'probot';
 import { inspect } from 'util';
+import { URL } from 'url';
```
Comment (erickzhao) [inline modules/bot/src/github-client.ts:4]: Is this addition necessary? 🤔

Comment (ckerr) [reply modules/bot/src/github-client.ts:4]: It shouldn't be, but unfortunately it is, for now.

Discussion @ https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34960

**Review by clavin — APPROVED** [review]
